Library/Homebrew/brew.sh

```diff
@@ -229,6 +234,51 @@ EOS
   fi
 fi
 
+# If HOMEBREW_FORCE_BREW_WRAPPER and HOMEBREW_DISABLE_NO_FORCE_BREW_WRAPPER are set,
+# verify that the path to our parent process is the same as the value of HOMEBREW_FORCE_BREW_WRAPPER for all
+# non-trivial commands (i.e. not defined above this line e.g. formulae or --cellar).
+if [[ -n "${HOMEBREW_FORCE_BREW_WRAPPER:-}" && -n "${HOMEBREW_DISABLE_NO_FORCE_BREW_WRAPPER:-}" ]]
+then
+  if [[ -n "${HOMEBREW_MACOS:-}" ]]
+  then
+    source "${HOMEBREW_LIBRARY}/Homebrew/utils/ruby.sh"
+    setup-ruby-path
+    HOMEBREW_BREW_CALLER="$("${HOMEBREW_RUBY_PATH}" "${HOMEBREW_LIBRARY}/Homebrew/utils/pid_path.rb" "${PPID}")"
+  else
+    HOMEBREW_BREW_CALLER="$(readlink -f "/proc/${PPID}/exe")"
+  fi
+  HOMEBREW_BREW_CALLER_CHECK_EXIT_CODE="$?"
+
+  if [[ "${HOMEBREW_BREW_CALLER_CHECK_EXIT_CODE}" -ne 0 ]]
+  then
+    # Error message already printed above when populating `HOMEBREW_BREW_CALLER`.
+    odie "failed to check the path to the parent process!"
+  fi
+
+  if [[ "${HOMEBREW_BREW_CALLER:-}" != "${HOMEBREW_FORCE_BREW_WRAPPER}" ]]
+  then
+    HOMEBREW_FORCE_BREW_WRAPPER_WITHOUT_BREW="${HOMEBREW_FORCE_BREW_WRAPPER%/brew}"
+
+    # HOMEBREW_ORIGINAL_BREW_FILE set by bin/brew
+    # shellcheck disable=SC2154
+    odie <<EOS
+conflicting Homebrew wrapper configuration!
+HOMEBREW_FORCE_BREW_WRAPPER was set to ${HOMEBREW_FORCE_BREW_WRAPPER}
+but \`brew\` was invoked by ${HOMEBREW_BREW_CALLER}.
+
+$(bold "Ensure you run ${HOMEBREW_FORCE_BREW_WRAPPER} directly (not ${HOMEBREW_ORIGINAL_BREW_FILE})")!
+
+Manually setting your PATH can interfere with Homebrew wrappers.
+Ensure your shell configuration contains:
+  eval "\$(${HOMEBREW_BREW_FILE} shellenv)"
+or that ${HOMEBREW_FORCE_BREW_WRAPPER_WITHOUT_BREW} comes before ${HOMEBREW_PREFIX}/bin in your PATH:
+  export PATH="${HOMEBREW_FORCE_BREW_WRAPPER_WITHOUT_BREW}:${HOMEBREW_PREFIX}/bin:\$PATH"
+EOS
+  fi
+
+  unset HOMEBREW_BREW_CALLER HOMEBREW_BREW_CALLER_CHECK_EXIT_CODE
+fi
+
```

Comment:
Would be good to DRY this up with changes above and/or migrate everything related to wrappers into into a new `utils/brew_wrapper.sh` or similar file instead.